src/common/dwt.c

```diff
@@ -380,33 +271,22 @@ static void dwt_wavelet_decompose(float *img, dwt_params_t *const p, _dwt_layer_
   int bcontinue = 1;
   const int size = p->width * p->height * p->ch;
 
+  assert(p->ch == 4);
+
   if(layer_func) layer_func(img, p, 0);
 
   if(p->scales <= 0) goto cleanup;
 
   /* image buffers */
   buffer[0] = img;
   /* temporary storage */
-  buffer[1] = dt_alloc_align(64, size * sizeof(float));
-  if(buffer[1] == NULL)
-  {
-    printf("not enough memory for wavelet decomposition");
-    goto cleanup;
-  }
-  memset(buffer[1], 0, size * sizeof(float));
-
-  // setup a temp buffer
-  temp = dt_alloc_align(64, MAX(p->width, p->height) * p->ch * sizeof(float));
-  if(temp == NULL)
-  {
-    printf("not enough memory for wavelet decomposition");
-    goto cleanup;
-  }
-  memset(temp, 0, MAX(p->width, p->height) * p->ch * sizeof(float));
-
+  buffer[1] = dt_alloc_sse_ps(size);
```

Comment:
it is a bit strange to see all sse code removed, but using dt_alloc_sse_ps() here.

Reply:
I used dt_alloc_sse_ps because it's specifically for allocating floats (so no need for sizeof(float)), and declares alignment to the compiler.  It is in fact just a wrapper for dt_alloc_align.  We could make a copy called something like dt_alloc_aligned_float to make the intent clearer.  And by hiding the alignment specification inside the function, it becomes a single location to update if we ever need to port to an architecture with cache lines bigger than 64 bytes.

Reply:
Yes, I'm for the copy with new name. My point is really about the code clarity even if current code is correct it is misleading to me.

Reply:
@ralfbrown : Ping ? You may have skipped this one ?

Reply:
@ralfbrown : I've done that and will merge.